chore(docs): organize installation doc

ncknt opened the pull request:
- group guides together
- present summary of installation methods
- shorter operator doc

Commit:
chore(docs): organize installation doc
- group guides together
- present summary of installation methods
- shorter operator doc

Commit:
update

dogonthehorizon commented on content/en/docs/installation/_index.md:
```diff
@@ -0,0 +1,38 @@
+| [Minnaker](./minnaker) | MacOS, Linux, Windows | Spin up a whole environment on top of Rancher to deploy Spinnaker | This is ideal if you do not have a Kubernetes cluster available |
+
+
+All the methods above share similar configurations and you can change between them.
```
Do we have something to guide users on how to move between configuration formats? Feels odd to call this out here without giving users a lifeline hah

ncknt replied:
Yes for operator. No for the other ones. I see what you're getting at but was hoping to keep that for a future iteration.

dogonthehorizon commented on content/en/docs/installation/armory-halyard.md:
```diff
@@ -0,0 +1,115 @@
+
+Running Armory Halyard in Docker is convenient and portable. The daemon will need access to files and environment variables, such as:
+- Halyard's main configuration directory - make sure the daemon has write access to that directory
+- kubeconfig file for Spinnaker's installation cluster (usually `~/.kube/config`)
```
Feel like we should either:
- Put this in backticks `kubeconfig`
- Capitalize it since it's the start of a line

dogonthehorizon commented on content/en/docs/installation/armory-halyard.md:
```diff
@@ -0,0 +1,115 @@
+- AWS profiles (usually `~/.aws`) if access to AWS is needed
+- Any other configuration files that reside on the Docker host
+
+> The Docker container expects to use `heptio-authenticator-aws` instead of `aws-iam-authenticator`
```
Is this still true? I don't even think you can get the heptio authenticator post-acquisition

ncknt replied:
I doubt it but I was just moving information around. A lot of stuff needs to be updated IMO.

dogonthehorizon commented on content/en/docs/installation/armory-halyard.md:
```diff
@@ -0,0 +1,115 @@
+
+## Run in Kubernetes
+
+Armory Halyard can also be installed as a Kubernetes `StatefulSet`. The advantage of running Halyard in the same cluster as Spinnaker is to get the same network access as Spinnaker itself in locked down some environments.
```
```suggestion
Armory Halyard can also be installed as a Kubernetes `StatefulSet`. The advantage of running Halyard in the same cluster as Spinnaker is to get the same network access as Spinnaker itself in some locked down environments.
```

dogonthehorizon commented on content/en/docs/installation/minnaker.md:
```diff
@@ -0,0 +1,17 @@
+
+
+### Prerequisites
+
```
RIP Oxford Comma :'(

ncknt replied:
I am a big fan of the Oxford comma. I, just moved, that paragraph, from, an existing doc.

dogonthehorizon commented on content/en/docs/installation/operator.md:
```diff
@@ -1,59 +1,22 @@
+- Store Spinnaker secrets in [Kubernetes secrets](https://github.com/armory/spinnaker-operator/blob/master/doc/managing-spinnaker.md#secrets-in-kubernetes-secrets).
+- Gain total control over Spinnaker manifests with [`kustomize` style patching](https://github.com/armory/spinnaker-operator/blob/master/doc/options.md#speckustomize)
 - Define Kubernetes accounts in `SpinnakerAccount` objects and store kubeconfig inline, in Kubernetes secrets, in s3, or GCS **(Experimental)**.
+- Deploy Spinnaker in an Istio controlled cluster **(Experimental)**
```
Would this list be better served in the product compatibility matrix? i.e. could we just link to our intro blog post here instead of listing all this out?

dogonthehorizon commented on content/en/docs/installation/operator.md:
```diff
@@ -62,51 +25,36 @@ Before you start, ensure the following requirements are met:
-
-### Download the Operator Manifests
+Operator has two distinct modes:
+- **Basic**: Operator in basic mode installs Spinnaker into a single namespace without `ValidatingAdmissionWebhook` for doing preflight checks.
```
```suggestion
- **Basic**: for installing the Operator and Spinnaker in a single namespace without doing pre-flight checks.
```

dogonthehorizon replied:
I don't think anyone reading this will care if it's a validating admissions webhook or not, just that we won't do pre-flight checks before applying the manifest. Thoughts?

dogonthehorizon commented on content/en/docs/installation/operator.md:
```diff
@@ -62,51 +25,36 @@ Before you start, ensure the following requirements are met:
-### Download the Operator Manifests
+Operator has two distinct modes:
+- **Basic**: Operator in basic mode installs Spinnaker into a single namespace without `ValidatingAdmissionWebhook` for doing preflight checks.
+- **Cluster**: Operator in cluster mode installs Spinnaker across namespaces with `ValidatingAdmissionWebhook` for doing preflight checks. This mode requires a `ClusterRole`.
```
```suggestion
- **Cluster**: which allows Operator to install Spinnaker across namespaces and provides pre-flight checks to prevent common misconfigurations. This mode requires a `ClusterRole`.
```

dogonthehorizon commented on content/en/docs/installation/operator.md:
```diff
@@ -116,125 +64,73 @@ NAMESPACE                             READY         STATUS       RESTARTS      A
-In the examples, the `spinnaker-namespace` parameter refers to the namespace where you want to install Spinnaker. It is likely different from Operator's namespace.
 
-**Important**: You must edit `deploy/spinnaker/basic/SpinnakerService.yml` to point to persistent storage, such as an S3 bucket. Other attributes can also be changed. For example, if you change the value of the `version` field to `2.16.0`, Operator installs version 2.16.0.
+See [the full format](../operator-reference)
```
Feels like we should link to the patch files repo so that users can see some common setups, in addition to the raw operator reference:
https://github.com/armory/spinnaker-kustomize-patches

dogonthehorizon commented on content/en/docs/installation/operator.md:
```diff
@@ -259,7 +155,105 @@ kubectl -n <namespace> describe spinnakerservice spinnaker
+There are many more possibilities such as:
+- managing manifests of MySQL instances
+- ensuring the same configuration is used between Staging and Production Spinnaker
+- splitting accounts in their own kustomization for an easy to maintain configuration 
```
This would also be a good section to call out the spinnaker-patch-files repo:
https://github.com/armory/spinnaker-kustomize-patches

dogonthehorizon commented on content/en/docs/installation/operator.md:
```diff
@@ -259,7 +155,105 @@ kubectl -n <namespace> describe spinnakerservice spinnaker
+
+### Secret Management
+
+You can store secrets in one of the [support secret engine](../../spinnaker-install-admin-guides/secrets/secrets/#supported-secret-engines). 
```
```suggestion
You can store secrets in one of the [supported secrets engines](../../spinnaker-install-admin-guides/secrets/secrets/#supported-secret-engines). 
```

dogonthehorizon commented on content/en/docs/installation/operator.md:
```diff
@@ -259,7 +155,105 @@ kubectl -n <namespace> describe spinnakerservice spinnaker
+
+You can store secrets in one of the [support secret engine](../../spinnaker-install-admin-guides/secrets/secrets/#supported-secret-engines). 
+
+#### Kubernetes Secret
```
```suggestion
#### Kubernetes Secrets
```

dogonthehorizon commented on content/en/docs/installation/operator.md:
```diff
@@ -259,7 +155,105 @@ kubectl -n <namespace> describe spinnakerservice spinnaker
+With the operator, you can also reference secrets stored in existing Kubernetes secrets in the same namespace as Spinnaker.
+
+The format is:
+- `encrypted:k8s!n:<secret name>!k:<secret key>` for string values. These will be added as environment variable to the Spinnaker deployment.
```
```suggestion
- `encrypted:k8s!n:<secret name>!k:<secret key>` for string values. This will be added as an environment variable to the Spinnaker deployment.
```

dogonthehorizon reviewed (commented):
Overall LGTM, some suggestions and content questions but otherwise is pretty solid. Thanks for taking the time to do this!

kevinawoo commented on content/en/docs/installation/operator.md:
```diff
@@ -259,7 +155,105 @@ kubectl -n <namespace> describe spinnakerservice spinnaker
+        - mountPath: /opt/spinnaker/config/halyard.yml
+          name: halconfig-volume
+          subPath: halyard-local.yml
+      - name: halyard
```
Curious, Halyard might need this to be set, i'm not 100% sure tho. I know we have it set in our [prod operator config](https://github.com/armory-io/armory-infrastructure/blob/master/spinnaker/prod-operator/deployment.yaml#L40-L42)

```yaml
          env:
          - name: SPRING_PROFILE_ACTIVE
            value: local
```

ncknt replied:
I don't believe so (I think local is standard but I'm not sure). This part is a carryover from the existing doc and I'll let others continue to improve and correct the doc once it's reorganized.

ncknt committed:
Update content/en/docs/installation/armory-halyard.md

Co-authored-by: Fernando Freire <fernando.freire@armory.io>

aimeeu commented:
@brian-armory @aimeeu 
- add `alias` entry in front matter for pages that have been moved or renamed
- add redirects to netlify.toml for deleted pages

ncknt committed:
Update content/en/docs/installation/operator.md

Co-authored-by: Fernando Freire <fernando.freire@armory.io>

ncknt committed:
Update content/en/docs/installation/operator.md

Co-authored-by: Fernando Freire <fernando.freire@armory.io>

ncknt committed:
Update content/en/docs/installation/operator.md

Co-authored-by: Fernando Freire <fernando.freire@armory.io>

317brian committed:
edits to the landing page

317brian committed:
copyedits

Commit:
Add redirects for changed links

Commit:
Merge branch 'doc/install' of github.com:armory/docs into doc/install

Commit:
Merge branch 'master' into doc/install

ncknt commented:
@aimeeu I've added redirects to the toml as best I could. What would be the point of aliases if we have redirects? Is this something you can help with?

aimeeu committed:
Add aliases to individual files that moved to a new section

Add aliases to armory-halyard-reference and operator since they
moved to new directories.

Add alias to docs/installation/_index.md for the deleted file
/docs/spinnaker/install.md

Remove armory-halyard redirect from netlify.toml

aimeeu commented:
> @aimeeu I've added redirects to the toml as best I could. What would be the point of aliases if we have redirects? Is this something you can help with?

@ncknt Thanks! I expected myself or Brian to take care of redirects. 

The front matter tag `aliases` is for page-level redirects (`redirect_from` in Jekyll ). If you move an entire section from one directory to another, configuring a section-level redirect in `netlify.toml` is better than adding aliases to each page.

I moved the armory-halyard -> armory-halyard-reference redirect out of `netlify.toml` and added an alias to armory-halyard-reference instead since it's only one page.
Same with operator.md -> added  the `- /docs/spinnaker/operator/` alias since the single file moved to a new directory.

For the deleted file `/docs/spinnaker/install.md`, I added a `/docs/spinnaker/install/` alias to /docs/installation/_index.md

I'll add a section to the docs team's Hugo page on when to add an alias vs when to put a redirect in `netlify.toml`.

aimeeu committed:
fix redirect (forgot /docs/)

aimeeu commented:
@brian-armory 
Page redirects tested:
- /docs/spinnaker/install/ -> redirects to docs/installation/
- /docs/spinnaker/operator/ -> redirects to docs/installation/operator/
- /docs/spinnaker/armory_halyard/ -> redirects to docs/installation/armory-halyard-reference/

Section redirects tested:
- /docs/spinnaker-install-admin-guides/installation/ -> redirects to docs/installation/guide/
- /docs/spinnaker-install-admin-guides/installation/air-gapped/ -> redirects to docs/installation/guide/air-gapped/
- /docs/operator-reference/ -> redirects to docs/installation/operator-reference/
- /docs/operator-reference/armory/ -> redirects to docs/installation/operator-reference/armory/

So LGTM

aimeeu committed:
Add description to installation pages

Add description to installation pages so they appear on the
section landing page.

aimeeu reviewed (approved):
LGTM

317brian commented:
Thanks again for all your work on this @ncknt! I'm going to run a link checker to verify the links are good and get this merged today.

317brian committed:
copyedits

317brian committed:
fix broken link

317brian reviewed (approved):
LGTM. Integrity, the link checker, returned only expected 404s, like example URLs.
